qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java

```diff
@@ -50,27 +51,33 @@ public static void waitForElasticsearch(String status, String index, Installatio
 
         // we loop here rather than letting httpclient handle retries so we can measure the entire waiting time
         final long startTime = System.currentTimeMillis();
+        long lastRequest = 0;
         long timeElapsed = 0;
         boolean started = false;
+        Throwable lastException = null;
         while (started == false && timeElapsed < waitTime) {
-            try {
-
-                final HttpResponse response = Request.Get("http://localhost:9200/_cluster/health")
-                    .connectTimeout((int) timeoutLength)
-                    .socketTimeout((int) timeoutLength)
-                    .execute()
-                    .returnResponse();
-
-                if (response.getStatusLine().getStatusCode() >= 300) {
-                    final String statusLine = response.getStatusLine().toString();
-                    final String body = EntityUtils.toString(response.getEntity());
-                    throw new RuntimeException("Connecting to elasticsearch cluster health API failed:\n" + statusLine+ "\n" + body);
-                }
+            if (System.currentTimeMillis() - lastRequest > requestInterval) {
+                try {
+
+                    final HttpResponse response = Request.Get("http://localhost:9200/_cluster/health")
+                        .connectTimeout((int) timeoutLength)
+                        .socketTimeout((int) timeoutLength)
+                        .execute()
+                        .returnResponse();
+
+                    if (response.getStatusLine().getStatusCode() >= 300) {
+                        final String statusLine = response.getStatusLine().toString();
+                        final String body = EntityUtils.toString(response.getEntity());
+                        throw new RuntimeException("Connecting to elasticsearch cluster health API failed:\n" + statusLine + "\n" + body);
+                    }
 
-                started = true;
+                    started = true;
 
-            } catch (IOException e) {
-                logger.info("Got exception when waiting for cluster health", e);
+                } catch (IOException e) {
+                    lastException = e;
```

Comment:
Instead of losing these completely, perhaps we should log at debug level?

Reply:
If we logged at debug, what would we have to do to get at that log output? Looks like right now, tests log only at `INFO` and above.

Reply:
Then let's suppress them? Take the first exception, suppress additional exceptions, and we will get all of them if we fail.

Reply:
I'm not sure I follow. What we are doing right now is suppressing all exceptions except for the most recent one before reaching the timeout. Are you suggesting we do the same except that we capture only the _first_ exception we encounter?

Reply:
I mean actually "suppressed exceptions":

```
IOException foundException;
...
} catch {IOException e) {
    if (foundException == null) {
        foundException = e;
    } else {
        foundException.addSuppressed(e);
    }   
}
```

Reply:
Ah. Honestly, I've never used suppressed exceptions before. Thanks for pointing this out. I've implemented as you described.